test/safprofile-test.js

```diff
@@ -0,0 +1,133 @@
+    const url = "/ZLUX/plugins/org.zowe.configjs/services/data/_current/"
+      + "org.zowe.zlux.sample.angular/user";
+  
+    const profile = "ZLUX.DEFAULT.CFG.ORG_ZOWE_ZLUX_SAMPLE_ANGULAR.GET.USER";
+    
+    const result = makeProfileNameForRequest(url, "GET", "DEFAULT");
+    assert(result === profile);
+  });
+  
+  it('should corectly generate service profiles with a long path', function() {
+    const url1 = "/ZLUX/plugins/org.zowe.zossystem.subsystems"
+        + "/services/data/_current/" + longStr1;
+    const url2 = url1 + "/" + longStr2;
+    const url3 = url2 + "/" + longStr3;
+    const url4 = url3 + "/" + longStr4;
+    const url5 = '/ZLUX/plugins/org.zowe.zlux.sample.angular/services/hello/'
+      + '_current/zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz'
+      + 'zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz'
+      + 'zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz'
+    
+    const profile1 = "ZLUX.DEFAULT.SVC.ORG_ZOWE_ZOSSYSTEM_SUBSYSTEMS.DATA.GET."
+        + longStr1;
+    const profile2 = profile1 + "." + longStr2;
+    const profile3 = "ZLUX.DEFAULT.SVC.ORG_ZOWE_ZOSSYSTEM_SUBSYSTEMS.DATA.GET."
+      + longStr1 + "." + longStr2;
+    const profile4 = profile3;
+    const profile5 = "ZLUX.0.SVC.ORG_ZOWE_ZLUX_SAMPLE_ANGULAR.HELLO.POST";
+    
+    
+    const result1 = makeProfileNameForRequest(url1, "GET", "DEFAULT");
+    console.log(result1, "result1.length", result1.length)
+    assert(result1 === profile1);
+    assert(result1.length <= safprofile.ZOWE_PROFILE_NAME_LEN);
+    
+    const result2 = makeProfileNameForRequest(url2, "GET", "DEFAULT");
+    console.log(result2, "result2.length", result2.length)
+    assert(result2 === profile2)
+    assert(result2.length <= safprofile.ZOWE_PROFILE_NAME_LEN);
+    
+    const result3 = makeProfileNameForRequest(url3, "GET", "DEFAULT");
+    console.log(result3, "result3.length", result3.length)
+    assert(result3 === profile3)
+    assert(result3.length <= safprofile.ZOWE_PROFILE_NAME_LEN);
+    
+    const result4 = makeProfileNameForRequest(url4, "GET", "DEFAULT");
+    console.log(result4, "result4.length", result4.length)
+    assert(result4 === profile4)
+    assert(result4.length <= safprofile.ZOWE_PROFILE_NAME_LEN);
+    
+    const result5 = makeProfileNameForRequest(url5, "POST", "0");
+    console.log(result5, "result5.length", result5.length)
+    assert(result5 === profile5)
+    assert(result5.length <= safprofile.ZOWE_PROFILE_NAME_LEN);
+  });
+  
+  it('should corectly generate config profiles with a long path', function() {
+    const url1 = "/ZLUX/plugins/org.zowe.configjs/services/data/_current/"
+      + "org.zowe.zlux.sample.angular/user" + longStr1;
+    const url2 = url1 + "/" + longStr2;
+    const url3 = url2 + "/" + longStr3;
+    const url4 = url3 + "/" + longStr4;
+    
+    const profile1 = "ZLUX.DEFAULT.CFG.ORG_ZOWE_ZLUX_SAMPLE_ANGULAR.GET.USER"
```

Comment:
DEFAULT here rather than 0

Reply:
It's explicitly passes "DEFAULT" in line 114, so that's correct. I don't think it matters what the particular value is, as long as they match.